test/acceptance/tests/connect/connect_inject_test.go

```diff
@@ -88,3 +92,88 @@ func TestConnectInject(t *testing.T) {
 		})
 	}
 }
+
+// Test the cleanup controller that cleans up force-killed pods.
+func TestConnectInject_CleanupController(t *testing.T) {
+	cases := []struct {
+		secure      bool
+		autoEncrypt bool
+	}{
+		{false, false},
+		{true, false},
+		{true, true},
+	}
+
+	for _, c := range cases {
+		name := fmt.Sprintf("secure: %t; auto-encrypt: %t", c.secure, c.autoEncrypt)
+		t.Run(name, func(t *testing.T) {
+			cfg := suite.Config()
+			ctx := suite.Environment().DefaultContext(t)
+
+			helmValues := map[string]string{
+				"connectInject.enabled":        "true",
+				"global.tls.enabled":           strconv.FormatBool(c.secure),
+				"global.tls.enableAutoEncrypt": strconv.FormatBool(c.autoEncrypt),
+				"global.acls.manageSystemACLs": strconv.FormatBool(c.secure),
+			}
+
+			releaseName := helpers.RandomName()
+			consulCluster := consul.NewHelmCluster(t, helmValues, ctx, cfg, releaseName)
+
+			consulCluster.Create(t)
+
+			logger.Log(t, "creating static-client deployment")
+			k8s.DeployKustomize(t, ctx.KubectlOptions(t), cfg.NoCleanupOnFailure, cfg.DebugDirectory, "../fixtures/cases/static-client-inject")
+
+			logger.Log(t, "waiting for static-client to be registered with Consul")
+			consulClient := consulCluster.SetupConsulClient(t, c.secure)
+			retry.Run(t, func(r *retry.R) {
+				for _, name := range []string{"static-client", "static-client-sidecar-proxy"} {
+					instances, _, err := consulClient.Catalog().Service(name, "", nil)
+					r.Check(err)
+
+					if len(instances) != 1 {
+						r.Errorf("expected 1 instance of %s", name)
+					}
+				}
+			})
+
+			ns := ctx.KubectlOptions(t).Namespace
+			pods, err := ctx.KubernetesClient(t).CoreV1().Pods(ns).List(context.Background(), metav1.ListOptions{LabelSelector: "app=static-client"})
+			require.NoError(t, err)
+
+			// There should only be one pod because we set replicas to 1 but
+			// in some cases, the previous test's static-client pod is still
+			// terminating so we filter by ready pods.
+			var podName string
+			for _, pod := range pods.Items {
+				if helpers.IsReady(pod) {
+					podName = pod.Name
+					break
+				}
```

Comment:
This can be removed once https://github.com/hashicorp/consul-helm/pull/818 is merged.